Add backup and restore guide (#194)
scalar-labs/scalar-kubernetes: docs/BackupRestoreGuide.md

> # Backup restore guide for Scalar products

Reviewer: Following the guidance in [Google Developer Documentation Style Guide ("Heading and title text")](https://developers.google.com/style/headings#heading-and-title-text), how about we use the following?:

```suggestion
# Back up and restore ScalarDB or ScalarDL data in a Kubernetes environment

> This guide explains how to backup/restore ScalarDB/ScalarDL data on the Kubernetes environment. We assume you use the managed databases provided by cloud services as a backend database of ScalarDB/ScalarDL. Specifically, we assume the following managed databases.

Reviewer: ```suggestion
This guide explains how to backup and restore ScalarDB or ScalarDL data in a Kubernetes environment. Please note that this guide assumes that you are using a managed database provided by a cloud service as the backend database for ScalarDB or ScalarDL. The following is a list of the managed databases that this guide assumes you might be using:
```

> * NoSQL (Not support transactions)
>    * Amazon DynamoDB
>    * Azure Cosmos DB for NoSQL
> * RDB (support transactions)
>    * Amazon RDS (MySQL, PostgreSQL, Oracle, and SQL Server)
>    * Amazon Aurora (MySQL and PostgreSQL)
>    * Azure Database (MySQL and PostgreSQL)

Reviewer: ```suggestion
* NoSQL (does not support transactions)
   * Amazon DynamoDB
   * Azure Cosmos DB for NoSQL
* RDB (supports transactions)
   * Amazon RDS
      * MySQL
      * Oracle
      * PostgreSQL
      * SQL Server
   * Amazon Aurora
      * MySQL
      * PostgreSQL
   * Azure Database
      * MySQL
      * PostgreSQL
```

> Please refer to the following document for more details on the ScalarDB's backup/restore in a transactionally-consistent way.
> 
> * [A Guide on How to Backup and Restore Databases Used Through ScalarDB](https://github.com/scalar-labs/scalardb/blob/master/docs/backup-restore.md)

Reviewer: ```suggestion
For details on how to back up and restore databases used with ScalarDB in a transactionally consistent way, see [A Guide on How to Backup and Restore Databases Used Through ScalarDB](https://github.com/scalar-labs/scalardb/blob/master/docs/backup-restore.md).
```

> ## Backup

Reviewer: ```suggestion
## Perform a back up
```

> ### What database do you use? How many databases do you use?

Reviewer: ```suggestion
### Confirm type of database and number of databases in use
```

> What operation you need depends on the kind of database (NoSQL or RDB) and the number of databases.

Reviewer: ```suggestion
How you perform backup and restore depends on the type of database (NoSQL or RDB) and the number of databases you are using.
```

> #### Using NoSQL or multiple databases

Reviewer: ```suggestion
#### NoSQL or multiple databases
```

> If you use a NoSQL database, please refer to the following document for more details on the backup operations.
> 
> * [Backup NoSQL database guide](./BackupNoSQL.md)
> 
> Also, if you use multiple databases under the Scalar products using [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md), you must do the backup operations according to the above document.

Reviewer: ```suggestion
If you are using a NoSQL database, or if you have two or more databases that are using the [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md) features, please see [Back up a NoSQL database in a Kubernetes environment](./BackupNoSQL.md) for details on how to perform a backup.
```

Author: As I mentioned in another comment, I think it would be better to describe `databases that are used by the [Multi-storage Transactions] or [Two-phase Commit Transactions]` instead of `databases that are using the [Multi-storage Transactions] or [Two-phase Commit Transactions]`.

What do you think?

Reviewer: I agree. I've reworded this part to match how we mention it in my earlier comment.

```suggestion
If you are using a NoSQL database, or if you have two or more databases that the [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md) feature uses, please see [Back up a NoSQL database in a Kubernetes environment](./BackupNoSQL.md) for details on how to perform a backup.
```

> #### Using single RDB

Reviewer: ```suggestion
#### Single RDB
```

> If you use a single RDB, please refer to the following document for more details on the backup operations.
> 
> * [Backup RDB guide](./BackupRDB.md)
> 
> Even if you use RDB, you must follow the [Backup NoSQL database guide](./BackupNoSQL.md) when you use more than two RDBs under the Scalar products using [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md).

Reviewer: ```suggestion
If you are using a single RDB, please see [Back up an RDB in a Kubernetes environment](./BackupRDB.md) for details on how to perform a backup.

If you have two or more RDBs that are using the [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md) feature, you must follow the instructions in [Back up a NoSQL database in a Kubernetes environment](./BackupNoSQL.md) instead.
```

Author: As I mentioned in another comment, I think it would be better to describe `RDBs that are used by the [Multi-storage Transactions] or [Two-phase Commit Transactions]` instead of `RDBs that are using the [Multi-storage Transactions] or [Two-phase Commit Transactions]`.

What do you think?

Reviewer: I agree. I've reworded this part to match how we mention it in my earlier comment.

```suggestion
If you are using a single RDB, please see [Back up an RDB in a Kubernetes environment](./BackupRDB.md) for details on how to perform a backup.

If you have two or more RDBs that the [Multi-storage Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/multi-storage-transactions.md) or [Two-phase Commit Transactions](https://github.com/scalar-labs/scalardb/blob/master/docs/two-phase-commit-transactions.md) feature uses, you must follow the instructions in [Back up a NoSQL database in a Kubernetes environment](./BackupNoSQL.md) instead.
```

> ## Restore

Reviewer: ```suggestion
## Restore data
```

> Please refer to the following document for the details on how to restore data in each managed database.
> 
> * [Restore database guide](./RestoreDatabase.md)

Reviewer: ```suggestion
For details on how to restore data in each managed database, please see [Restore a database in a Kubernetes environment](./RestoreDatabase.md).
```
